### DISCUSSION TITLE
Artifact update problem

### DISCUSSION THREAD

**Discussion by julienmonnard** [renovatebot/renovate#10785] [Request Help]
Hello! We are using Renovate in our project for some time, but it was a bit left to the side and I'm trying to clean it up now. We are apparently facing an issue for some time, and I have no idea how to fix it. Basically for every PR that Renovate creates, it doesn't manage to update the package-lock.json file.

I am not sure if it's on our side or on the Renovate side, but since we are NOT self-hosting, I assume it's Renovate.

Here is a screen with the beginning of the log
![image](https://user-images.githubusercontent.com/29894873/125253709-fcbe5c00-e2f9-11eb-9abb-702a202fad6e.png)

I didn't want to just copy the log because it's the same line printed for every dependency until the max amount of characters is reached.

Is there something we can do about this? I can try to give more details if needed, let me know.

Thank you

**Comment by rarkins** [comment]
Does it happen for every repository, or just one?

Is it possible that the repo is very large and Renovate has run out of space?

Do you have any `.npmrc` file in the repo or `npmrc` settings in your Renovate config?

**Reply by julienmonnard** [reply]
Hey, thanks for the fast reply!

I saw it only on this repository, the other ones seem fine and we do not have any `.npmrc` or `npmrc` settings

About running out of space I guess it could be possible, we have quite a lot of dependencies. The first messages like this started to appear 22nd of April 2021. So it could be just some cache stacking up and using all the space.

Like I said, all the renovate PRs were not taken care of in a while, almost a year, so I might be miss something here...

If it is indeed lack of available space, is there a way to free up some?

Thanks again for your help!

**Reply by julienmonnard** [reply]
@rarkins I notice now that the 22nd of April is also the day v25 was released. Maybe there is a connection? Also 3 days ago I nuked the config and made a new onboarding, but it didn't solve the problem

**Reply by rarkins** [reply]
The hosted app didn't deploy v25 for a few weeks after release so that's probably unrelated

**Reply by julienmonnard** [reply]
Ok I see. Is there a way for you or me to see the available space for this repo? Or what could we do to fix this?

**Reply by rarkins** [reply]
I'm not sure yet if it's definitely disk related. What's a recent job ID?

**Reply by julienmonnard** [reply]
The latest Job ID, from about an hour ago is 401202019

**Reply by rarkins** [reply]
I've switched this repo's disk to be a larger one but I'm not sure that's the root cause. There seem to be many causes for `WARN tar ENOENT: no such file or directory` errors in npm if you search Google or Stack Overflow.

**Reply by julienmonnard** [reply]
Ok, so I tried to get a PR created by Renovate today and it's still having the problem, so the disk space might indeed not be the problem. I checked the latest logs available in the dashboard and I saw that this command seems to be failing, could it help with the investigation? Here it is from the job ID 404707039 but it appears multiple times in the logs of every job

![image](https://user-images.githubusercontent.com/29894873/126070200-f4912748-7c25-4271-b659-c660eca40f1d.png)

**Reply by rarkins** [reply]
Unfortunately all that means is that `npm install` is failing

**Reply by julienmonnard** [reply]
Ok. But since we are not hosting this, I guess there is not much I can do to fix it from my side.

I would like to use this tool again but it's not really possible for now, because the only way to make it work currently is to manually commit the updated lock file on each PR...

What would be my options now to make it work?

**Reply by rarkins** [reply]
One thing you could do is delete `node_modules` locally and then see if `npm install --no-audit --ignore-scripts` works or not on your base directory. Maybe it's the script part causing the problem.

You could also Google on this error (`WARN tar ENOENT: no such file or directory`) to see if there are any common causes or solutions which might be applicable.

**Reply by julienmonnard** [reply]
I spent some more time investigating this. Apparently when we first introduced Renovate to our process, the private repositories were not blocking, and we just didn't configure them. Now it's different and the artifacts are not updated without private packages config, even if it's not a private one being updated.
Now I added the full config for private repos and it seems to be working again.

Thanks for your help :)
